tasks/Debian.yml

```diff
@@ -1,36 +1,31 @@
-- name: Get uname
-  command: uname -r
-  register: os_uname
-
-# would rather have used ansible apt-key...
-- name: Add specific key
-  command: apt-key adv --keyserver keyserver.ubuntu.com --recv-keys {{docker_repo_key}}
-
-- name: Add docker repo
-  command: sudo sh -c "echo deb {{docker_repo}} docker main > /etc/apt/sources.list.d/docker.list"
-
-- name: Install os packages
-  apt: pkg={{item}} state=present update_cache=yes
-  with_items: 
-    - linux-image-extra-{{os_uname.stdout}}
-    - lxc-docker
-
-- name: Docker default config file
-  template: src=docker-defaults.j2 dest=/etc/default/docker
-
-- name: Docker init file
-  template: src=docker-init.j2 dest=/etc/init/docker.conf
-  notify: Restart Docker
-
-# consider seperate role here
-- name: Change ufw forward policy to ACCEPT
-  command: sed -i 's/DEFAULT_FORWARD_POLICY="DROP"/DEFAULT_FORWARD_POLICY="ACCEPT"/g' /etc/default/ufw
-  when: docker_listen_tcp == True
-
-- name: Reload ufw
-  command: ufw reload
-  when: docker_listen_tcp == True
-
-- name: Allow incomming tcp traffic on {{docker_listen_port}}
-  command: ufw allow {{docker_listen_port}}/tcp
-  when: docker_listen_tcp == True
+- name: "Import Docker APT public key."
+  apt_key:
+    keyserver: "{{ docker_pubkey_server }}"
+    id: "{{ docker_pubkey_id }}"
+    state: present
+  tags:
+    - docker
+    - package
+
+- name: "Install apt-transport-https if necessary."
+  apt:
+    name: "apt-transport-https"
+    state: present
+  tags:
+    - docker
+    - package
+
+- name: "Add the APT Docker repository."
+  apt_repository:
+    repo: "{{ docker_apt_repo }}"
+    state: present
+    filename: 'docker'
+  tags:
+    - docker
+    - package
+
+- name: "Install Docker"
+  apt: name=docker-engine state=present update_cache=yes
```

Comment:
`linux-image-extra-virtual` should also be installed to get some nice stuff.

The `update_cache` should not be required. `apt_repository` is supposed to take care of it.